ert3/storage/_storage.py

```diff
@@ -356,13 +398,58 @@ def get_ensemble_record(
 ) -> ert3.data.EnsembleRecord:
     if experiment_name is None:
         experiment_name = f"{workspace}.{_ENSEMBLE_RECORDS}"
+    experiment = _get_experiment_by_name(experiment_name)
+    if experiment is None:
+        raise ert3.exceptions.NonExistantExperiment(
+            f"Cannot get {record_name} data, no experiment named: {experiment_name}"
+        )
 
-    return _get_numerical_data(workspace, experiment_name, record_name)
+    param_names = _get_experiment_parameters(workspace, experiment_name)
+    if record_name in param_names:
```

Comment:
I think it would make sense to separate this entire `if`-clause into a separate function...

Reply:
I separated only the combining part.